### PR TITLE
chore: full repo structure and code

### PR DESCRIPTION
**Pull request by lucasfcosta**
As we discussed on chaijs/check-error#11 and chaijs/chai#817 here goes the whole code for splitting `getFunctionName` into a new module.

Please notice that we need saucelabs tokens here too in order to build and release this.

As soon as this gets merged and release I will refactor `check-error` and `chai` to use this.

### CODE REVIEW

**Comment by meeber** [comment]
@lucasfcosta My hero.

One question: What's the `test.sh`?

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,506 @@
+#!/usr/bin/env bash
```
Comment (keithamus) [inline test.sh]: I think you accidentally a file.

Comment (lucasfcosta) [inline test.sh]: Ah, thanks, that was the file I was using to install IE vms and test on them.

**Comment by keithamus** [comment]
Nitpick on commit message btw: I think it should be `feat: initial implementation`.

**Comment by lucasfcosta** [comment]
Here it is!
I renamed the commit and removed that file.
Please let me know if I missed anything.

**Comment by meeber** [comment]
LGTM

**Pull request merged**